scipy/sparse/_compressed.py

```diff
@@ -1464,6 +1464,42 @@ def _broadcast_to(self, shape, copy=False):
             indices = np.tile(np.arange(N), len(self.data))
             indptr = self.indptr * N
         return self.__class__((data, indices, indptr), shape=shape, copy=False)
+    
+    def __binsparse_descriptor__(self) -> dict:
+        """Return a `dict` equivalent to a parsed JSON [`binsparse` descriptor](https://graphblas.org/binsparse-specification/#descriptor)
+        of this array.
+
+        Returns
+        -------
+        dict
+            Parsed `binsparse` descriptor.
+        """
+        from scipy import __version__
+
+        data_dt = str(self.data.dtype)
+        if np.issubdtype(data_dt, np.complexfloating):
+            data_dt = f"complex[float{self.data.dtype.itemsize * 4}]"
+        return {
+            "binsparse": {
+                "version": "0.1",
+                "format": self.format.upper(),
+                "shape": list(self.shape),
+                "number_of_stored_values": self.nnz,
+                "data_types": {
+                    "pointers_to_1": str(self.indices.dtype),
+                    "indices_1": str(self.indptr.dtype),
+                    "values": data_dt,
+                },
+            },
+            "original_source": f"`scipy.sparse`, version {__version__}",
```

Comment:
This is a bit questionable, it makes more sense for a stored file on disk rather than an in-memory format. If it's left out then you'd not have to do the `desc = desc["binsparse"]` unpacking.

Reply:
I was following the structure in the [specification](https://graphblas.org/binsparse-specification/#descriptor) as closely as possible. Of course, one could change https://github.com/data-apis/array-api/issues/840 to say that we only return the `binsparse` key of the descriptor, and then this (and the unpacking) would go away.